hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java

```diff
@@ -290,6 +248,54 @@ protected boolean tryToExecuteSchedule(
         }
     }
 
+    protected boolean validationOk(final ResponseCodeEnum validationResult) {
+        return validationResult == ResponseCodeEnum.OK
+                || validationResult == ResponseCodeEnum.SUCCESS
+                || validationResult == ResponseCodeEnum.SCHEDULE_PENDING_EXPIRATION;
+    }
+
+    @NonNull
+    private Set<Key> getKeySetFromTransactionKeys(final TransactionKeys requiredKeys) {
+        final Set<Key> scheduledRequiredKeys = new ConcurrentSkipListSet<>(new RecordIdentityComparator());
+        scheduledRequiredKeys.addAll(requiredKeys.requiredNonPayerKeys());
+        scheduledRequiredKeys.addAll(requiredKeys.optionalNonPayerKeys());
+        scheduledRequiredKeys.add(requiredKeys.payerKey());
+        return scheduledRequiredKeys;
+    }
+
+    /**
+     * Given an arbitrary {@link Iterable<Key>}, return a <strong>modifiable</strong> {@link Set<Key>} containing
+     * the same objects as the input.
+     * If there are any duplicates in the input, only one of each will be in the result.
+     * If there are any null values in the input, those values will be excluded from the result.
+     * Note: The return value of this method is always safe to be read and written by multiple threads
+     *     concurrently, and operates as close to lock-free as possible.
```

Comment:
If this is true, seems like a return value of an interface other than `Set<Key>` would make more sense. Btw I realize this method was only moved, so not a blocker

Reply:
The note is followed by a comment to future implementors, is currently true, and must be maintained.
The use of `Set` is because only very specific implementation classes currently meet the requirement, and there is no common superclass or interface besides `Set` for them (basically Set doesn't have sub-interfaces, and the API design for the concurrent versions was a bit half-baked).
The alternative would be to have the method declare a particular implementation class, which is not good design.